packages/dds/tree/src/tree/types.ts

```diff
@@ -0,0 +1,7 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export type TreeKey = (number | string) & { readonly TreeKey: symbol; };
```

Comment:
I'm not clear on how separate we want to keep our prototypes, but the schema prototype does have types you could use for these. Particularly TreeType could be be set to TreeSchemaIdentifier here.

TreeKey is more of a challenge since the schema prototype assume local and global keys are in separate namespaces so `LocalFieldKey | GlobalFieldKey` isn't quite usable unless we reserve a GlobalFieldKey prefix or something like that.

Reply:
If you do want to share types, its probably better to move the types from schema here, then depend on them from schema as that lines up with the dependencies we want.

Reply:
Good point.  Let's not do it in this PR, but I think tackling defining our 'Id' and 'Key' type(s) should be the next thing we tackled to unblock coding.